azure/scope/cluster.go

```diff
@@ -193,15 +196,16 @@ func (s *ClusterScope) PublicIPSpecs() []azure.ResourceSpecGetter {
 	for _, subnet := range s.NodeSubnets() {
 		if subnet.IsNatGatewayEnabled() {
 			nodeNatGatewayIPSpecs = append(nodeNatGatewayIPSpecs, &publicips.PublicIPSpec{
-				Name:           subnet.NatGateway.NatGatewayIP.Name,
-				ResourceGroup:  s.ResourceGroup(),
-				DNSName:        subnet.NatGateway.NatGatewayIP.DNSName,
-				IsIPv6:         false, // Public IP is IPv4 by default
-				ClusterName:    s.ClusterName(),
-				Location:       s.Location(),
-				FailureDomains: s.FailureDomains(),
-				AdditionalTags: s.AdditionalTags(),
-				IPTags:         subnet.NatGateway.NatGatewayIP.IPTags,
+				Name:             subnet.NatGateway.NatGatewayIP.Name,
+				ResourceGroup:    s.ResourceGroup(),
+				DNSName:          subnet.NatGateway.NatGatewayIP.DNSName,
+				IsIPv6:           false, // Public IP is IPv4 by default
+				ClusterName:      s.ClusterName(),
+				Location:         s.Location(),
+				ExtendedLocation: s.ExtendedLocation(),
```

Comment:
Do we need to add ExtendedLocation here? 
As per the issue it is mentioned that public MEC don't have NAT gateway

Reply:
Will remove ExtendedLocation in nodeNatGateway here, also in azureBastion as public MEC has not supported it.